test/features/environment.py

```diff
@@ -0,0 +1,25 @@
+import logging
+import re
+from crmsh import utils, parallax
+
+
+def get_online_nodes(context):
+    rc, out, err = utils.get_stdout_stderr('crm_node -l')
+    if rc != 0 and err:
+        context.logger.error("{}\n".format(err))
+        return []
+    return re.findall(r'[0-9]+ (.*) member', out)
+
+
+def before_tag(context, tag):
+    context.logger = logging.getLogger("test.{}".format(tag))
+    # tag @clean means need to stop cluster service
+    if tag == "clean":
+        online_nodes = get_online_nodes(context)
+        if online_nodes:
+            try:
+                parallax.parallax_call(online_nodes, 'crm cluster stop')
+            except ValueError as err:
+                context.logger.error("{}\n".format(err))
```

Comment:
Quick question. What happens here if we have a failure? Are tests executed?
In this exception we should finish the test execution, right?

Reply:
changed, set `context.failed = True` maybe work